controller/cache/info.go

```diff
@@ -170,6 +191,45 @@ func populateIngressInfo(un *unstructured.Unstructured, res *ResourceInfo) {
 	res.NetworkingInfo = &v1alpha1.ResourceNetworkingInfo{TargetRefs: targets, Ingress: ingress, ExternalURLs: urls}
 }
 
+func populateRouteInfo(un *unstructured.Unstructured, res *ResourceInfo) {
+	route := getRoute(un)
+	targetsMap := make(map[v1alpha1.ResourceRef]bool)
+
+	if backend, ok, err := unstructured.NestedMap(un.Object, "spec", "to"); ok && err == nil {
+		targetsMap[v1alpha1.ResourceRef{
+			Kind:      kube.ServiceKind,
+			Namespace: un.GetNamespace(),
+			Name:      fmt.Sprintf("%s", backend["name"]),
+		}] = true
+	}
+
+	targets := make([]v1alpha1.ResourceRef, 0)
+	for target := range targetsMap {
+		targets = append(targets, target)
+	}
+
+	var urls []string
+
+	if hostName, ok, err := unstructured.NestedString(un.Object, "spec", "host"); ok && err == nil {
+		stringPort := "http"
+		if tls, _, err := unstructured.NestedMap(un.Object, "spec", "tls"); err == nil && tls != nil {
+			stringPort = "https"
+		}
+		url := fmt.Sprintf("%s://%s", stringPort, hostName)
+		if path, ok, err := unstructured.NestedString(un.Object, "spec", "path"); ok && err == nil {
+			path = strings.TrimSuffix(path, "*")
+			url += path
+		}
+		if res.NetworkingInfo != nil {
+			urls = res.NetworkingInfo.ExternalURLs
+		}
+		urls = append(urls, url)
+	}
+
+	res.NetworkingInfo = &v1alpha1.ResourceNetworkingInfo{TargetRefs: targets, Ingress: route, ExternalURLs: urls}
```

Comment:
I think we need to handle the edge cases when we are unable to extract backend services, ingress, and urls.